pkg/streamingpromql/types/data.go

```diff
@@ -102,3 +105,36 @@ type ScalarData struct {
 	// Samples must not have duplicate timestamps.
 	Samples []promql.FPoint
 }
+
+func HasDuplicateSeries(metadata []SeriesMetadata) (bool, labels.Labels) {
+	switch len(metadata) {
+	case 0, 1:
+		return false, labels.EmptyLabels()
+	case 2:
+		if metadata[0].Labels.Hash() == metadata[1].Labels.Hash() {
+			return true, metadata[0].Labels
+		}
+
+		return false, labels.EmptyLabels()
+
+	default:
+		// Note that there's a risk here that we incorrectly flag two series as duplicates when in reality they're different
+		// due to hash collisions.
```

Comment:
(nit) this applies to case 2 above as well. Perhaps move this there?

Reply:
Done in 0cc5310